# NOTEBOOK #01 - Inspección Inicial

**Proyecto Integrador - Minería de Datos I - Tec. Sup. Ciencia de Datos - ITSE**  
**Integrantes: Daniela Fernández - Julio Nahuel Gomez**

**Dataset:** Usuarios de plataforma de streaming  
**Objetivo:** Explorar la estructura general del dataset, identificar tipos de datos, valores faltantes, duplicados e inconsistencias, sin realizar ninguna modificación sobre los datos originales.


Primero importamos las librerías que vamos a usar. pandas para manejar la tabla de datos, numpy para operaciones numéricas, y matplotlib y seaborn para gráficos.

In [14]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns


Segundo cargamos el archivo CSV en una variable llamada df (de DataFrame, que es como una tabla en pandas) que nos mostrará las primeras 5 filas para ver cómo lucen los datos.

In [16]:
# Carga del dataset
df = pd.read_json('streaming_users_dirty.csv')

# Ver las primeras 5 filas
df.head()

,user_id,age,subscription_plan,monthly_watch_time_mins,country,favorite_genre,last_login_date,customer_support_tickets
0,10000,39,Estándar,805.8,Brasil,Crime,2025-03-04,99
1,10001,37,Estándar,1173.4,Colombia,Crime,2019-04-02,2
2,10002,28,Básico,401.0,Colombia,Crime,2018-04-13,0
3,10003,43,Básico,62.4,Uruguay,Thriller,2021-01-31,0
4,10004,51,Básico,477.8,Perú,Thriller,2020-09-30,1


## 1. Estructura general del dataset

In [5]:
# Dimensiones del dataset
#En esta celda nos dice cuántas filas y columnas tiene el dataset, y los nombres de cada columna.

print(f"Filas: {df.shape[0]}")
print(f"Columnas: {df.shape[1]}")
print(f"\nNombre de las columnas: {list(df.columns)}")

Filas: 8160
Columnas: 8

Nombre de las columnas: ['user_id', 'age', 'subscription_plan', 'monthly_watch_time_mins', 'country', 'favorite_genre', 'last_login_date', 'customer_support_tickets']


In [6]:
# Tipos de datos de cada columna, por ej. si es un numero entero (int64), número decimal (float64), texto (object) o fecha.
print("Tipos de datos:")
print(df.dtypes)

Tipos de datos:
user_id                       int64
age                           int64
subscription_plan            object
monthly_watch_time_mins     float64
country                      object
favorite_genre               object
last_login_date              object
customer_support_tickets      int64
dtype: object


En la celda de arriba observamos que:
* user_id → int64 (número entero)  
* age → int64 (número entero)
* subscription_plan → object (texto)
* monthly_watch_time_mins → float64 (decimal)
* country → object (texto)
* favorite_genre → object (texto)
* last_login_date → object (texto) (en este caso figura como object (texto) cuando debería ser de tipo fecha)
* customer_support_tickets → int64 (número entero)

In [9]:
# Valores faltantes por columna: contamos cuántos valores nulos (vacíos) hay en cada columna.
#Un valor faltante significa que ese campo no tiene dato para ese registro.

print("Valores faltantes por columna:")
print(df.isnull().sum())
print(f"\nTotal de valores faltantes: {df.isnull().sum().sum()}")

Valores faltantes por columna:
user_id                       0
age                           0
subscription_plan             0
monthly_watch_time_mins     193
country                       0
favorite_genre              240
last_login_date             320
customer_support_tickets      0
dtype: int64

Total de valores faltantes: 753


En este caso encontramos 753 valores faltantes en total distribuidos así:

* monthly_watch_time_mins → 193 nulos
* favorite_genre → 240 nulos
* last_login_date → 320 nulos

In [10]:
# Registros duplicados: Verificamos si hay filas exactamente iguales en el dataset.
#Un duplicado significa que un mismo usuario aparece más de una vez con exactamente los mismos datos.

duplicados = df.duplicated().sum()
print(f"Registros duplicados: {duplicados}")

Registros duplicados: 126


Hay 126 registros duplicados. Eso significa que 126 filas aparecen más de una vez con exactamente los mismos datos.

In [11]:
# Valores únicos por columna categórica
print("Valores únicos en subscription_plan:")
print(df['subscription_plan'].unique())

print("\nValores únicos en country:")
print(df['country'].unique())

print("\nValores únicos en favorite_genre:")
print(df['favorite_genre'].unique())

Valores únicos en subscription_plan:
['Estándar' 'Básico' 'Premium' 'Std' 'estandar' 'basico' 'básico'
 'Premium ' 'premium' 'Premiun' 'BASICO' 'STANDARD' 'Basic' 'Estándar '
 'PREMIUM']

Valores únicos en country:
['Brasil' 'Colombia' 'Uruguay' 'Perú' 'Chile' 'Argentina' 'México'
 'Brazil' 'brasil' 'méxico' 'chile' 'uruguay' 'MEX' 'ARG' 'colombia' 'COL'
 'Mexico' 'URY' 'Chile ' 'Peru' 'perú' 'argentina' 'PER' 'Argentina '
 'BRA' 'CHL']

Valores únicos en favorite_genre:
['Crime' 'Thriller' 'Drama' 'Acción' 'Romance' 'Comedia' 'Documental'
 'ACCIÓN' None 'CRIME' 'Comedia ' 'comedy' 'DRAMA' 'THRILLER'
 'Documentary' 'Thriller ' 'accion' 'Action' 'COMEDIA' 'ROMANCE' 'thriler'
 'DOC' 'romance' 'Crimen' 'Drama ' 'drama' 'Romance ' 'documental' 'crime']


En la celda de arriba nos muestra todos los valores distintos que aparecen en las columnas de texto. Esto nos permite detectar inconsistencias como un mismo valor escrito de distintas formas, por ejemplo "Premium", "PREMIUM", "premium", que deberían ser el mismo.

En el código podemos observar que:
* subscription_plan tiene 15 variantes para solo 3 planes: Estándar, Básico y Premium escritos de formas distintas (mayúsculas, minúsculas, con acento, sin acento, abreviados).
* country tiene los países escritos con nombres completos, abreviados (MEX, ARG, BRA...), en minúsculas, con espacios extra.
* favorite_genre tiene géneros en español e inglés mezclados, en mayúsculas, minúsculas, con errores tipográficos (thriler, accion) y hasta None.

In [12]:
# Estadísticas básicas de columnas numéricas

print("Estadísticas descriptivas de columnas numéricas:")
df[['age', 'monthly_watch_time_mins', 'customer_support_tickets']].describe()

Estadísticas descriptivas de columnas numéricas:


,age,monthly_watch_time_mins,customer_support_tickets
count,8160.000000,7967.000000,8160.000000
mean,34.096814,1107.346153,1.800980
std,14.511304,5310.442604,11.334969
min,-5.000000,-120.000000,-1.000000
25%,25.000000,489.200000,0.000000
50%,33.000000,757.400000,1.000000
75%,42.000000,1045.700000,1.000000
max,150.000000,99999.000000,150.000000


En este caso nos muestra el mínimo, máximo, promedio, mediana y otros estadísticos de las columnas numéricas. Nos permite detectar valores imposibles como edades negativas o mayores a 120.
Ademas podemos observar que se confirman varios problemas graves en los datos numéricos:
* age: mínimo -5 y máximo 150 — ambos son imposibles para una edad real.
* monthly_watch_time_mins: mínimo -120 (tiempo negativo, imposible) y máximo 99999 (más de 69 días seguidos viendo contenido, claramente un error).
* customer_support_tickets: mínimo -1 (tickets negativos, imposible) y máximo 150 (valor extremadamente alto).

## Observaciones finales de la inspección inicial

A partir de la inspección que realizamos identificamos los siguientes problemas en el dataset:

- **Dimensiones:** 8.160 filas y 8 columnas.
- **Valores faltantes:** 753 en total. Las columnas afectadas son `monthly_watch_time_mins` (193), `favorite_genre` (240) y `last_login_date` (320).
- **Duplicados:** 126 registros duplicados.
- **Tipo de dato incorrecto:** `last_login_date` figura como texto (object) cuando debería ser de tipo fecha.
- **Inconsistencias en variables categóricas:** `subscription_plan`, `country` y `favorite_genre` presentan múltiples variantes para un mismo valor (mayúsculas, minúsculas, abreviaciones, errores tipográficos).
- **Valores imposibles en variables numéricas:** edades negativas y mayores a 120, tiempos de visualización negativos y extremadamente altos, tickets de soporte negativos.
